chanzuckerberg/corpora-data-portal
# 401-landing-page

This PR aims to create a temporary landing page for Data Portal as described in:

https://www.figma.com/proto/Ys3WOryh4RpT1V5ezlOtEL/M1-Find-Data-explorations?node-id=158%3A2761&scaling=min-zoom

https://www.figma.com/file/Ys3WOryh4RpT1V5ezlOtEL/M1-Find-Data-explorations?node-id=158%3A2441

Changes:
1. Add TS definitions for schemas and entities used to create the page. Such as Project, Dataset, DatasetAsset, etc.. Note that there are properties intentionally commented out in the type definitions, since they're not being used in the FE yet. I'm assuming as we build more features, we can uncomment the properties as they start being used in the app. 

NOTE: I followed `backend/config/corpora-api.yml` to build the types

1. Add `Styled Components` for styling for now. Given that branding / design system discussion is still ongoing, I think it's too early to commit to a certain styling library, so I just picked one that I'm most familiar with 😄 

1. Add lint rule `sort-keys`, so files have consistent object sort order

1. Add hard coded `PROJECTS` data to showcase the UI, this will be removed once we integrate FE with BE API (https://github.com/chanzuckerberg/corpora-data-portal/pull/433)

1. Update `frontend/src/pages/index.tsx` to use the new components to render the landing page

1. Add `.codeclimate.yml` to ignore files

QA:

1. To start the app, navigate to `corpora-data-portal/frontend` folder
2. run `npm i && npm run develop`
3. Visit http://localhost:8000/ to see the landing page

NOTE: Clicking on any cellxgene links won't work right now, because it's hard coded data. This should work once API integration is done!

Thank you for your reviews 😄 🙏 !

<img width="1680" alt="Screen Shot 2020-07-14 at 9 16 04 AM" src="https://user-images.githubusercontent.com/6309723/87452386-09f51d80-c5b6-11ea-88ea-02da194fccf7.png">

<img width="1271" alt="Screen Shot 2020-07-14 at 9 16 16 AM" src="https://user-images.githubusercontent.com/6309723/87452393-0b264a80-c5b6-11ea-92ff-034d240acc6f.png">

![gif](https://user-images.githubusercontent.com/6309723/87452378-05c90000-c5b6-11ea-8d76-e802e847bb43.gif)

closes #401
closes #312
